mods/letsgo/rulesets.js

```diff
@@ -0,0 +1,171 @@
+				let item = this.getItem(set.item);
+				if (item.megaEvolves && item.megaEvolves !== template.baseSpecies) set.item = '';
+			}
+
+			// Legendary Pokemon must have at least 3 perfect IVs in gen 6
+			if (set.ivs && this.gen >= 6 && (baseTemplate.gen >= 6 || format.requirePentagon) && (template.eggGroups[0] === 'Undiscovered' || template.species === 'Manaphy') && !template.prevo && !template.nfe &&
+				// exceptions
+				template.species !== 'Unown' && template.baseSpecies !== 'Pikachu' && (template.baseSpecies !== 'Diancie' || !set.shiny)) {
+				let perfectIVs = 0;
+				for (let i in set.ivs) {
+					// @ts-ignore
+					if (set.ivs[i] >= 31) perfectIVs++;
+				}
+				let reason = (format.requirePentagon ? " and this format requires gen " + this.gen + " Pokémon" : " in gen 6");
+				if (perfectIVs < 3) problems.push((set.name || set.species) + " must have at least three perfect IVs because it's a legendary" + reason + ".");
+			}
+
+			// limit one of each move
+			let moves = [];
+			if (set.moves) {
+				/**@type {{[k: string]: true}} */
+				let hasMove = {};
+				for (const moveId of set.moves) {
+					let move = this.getMove(moveId);
+					let moveid = move.id;
+					if (hasMove[moveid]) continue;
+					hasMove[moveid] = true;
+					moves.push(moveId);
+				}
+			}
+			set.moves = moves;
+
+			let battleForme = template.battleOnly && template.species;
+			if (battleForme) {
+				if (template.isMega) set.species = template.baseSpecies;
+				if (template.requiredMove && !set.moves.includes(toId(template.requiredMove))) {
+					problems.push(`${template.species} transforms in-battle with ${template.requiredMove}.`); // Meloetta-Pirouette, Rayquaza-Mega
+				}
+			} else {
+				if (template.requiredMove && !set.moves.includes(toId(template.requiredMove))) {
+					problems.push(`${(set.name || set.species)} needs to have the move ${template.requiredMove}.`); // Keldeo-Resolute
+				}
+			}
+
+			return problems;
+		},
+	},
+	allowavs: {
+		effectType: 'ValidatorRule',
+		name: 'Allow AVs',
+		desc: "Tells formats with the 'letsgo' mod to take Awakening Values into consideration when calculating stats",
+		onChangeSet: function (set, format) {
+			/**@type {string[]} */
+			let problems = ([]);
+			let totalAV = 0;
+			if (set.evs) {
+				for (let k in set.evs) {
+					let av = this.getAwakeningValues(set);
+					// @ts-ignore
+					av[k] = set.evs[k];
+					// @ts-ignore
+					if (typeof av[k] !== 'number' || av[k] < 0) {
+						// @ts-ignore
+						av[k] = 0;
+					}
+					// @ts-ignore
+					totalAV += av[k];
+				}
+			}
+
+			// ----------- legality line ------------------------------------------
+			if (!this.getRuleTable(format).has('-illegal')) return problems;
+			// everything after this line only happens if we're doing legality enforcement
+
+			// Pokemon cannot have more than 1200 total Awakening Values
+			if (totalAV > 1200) {
+				problems.push((set.name || set.species) + " has more than 1200 total Awakening Values.");
```

Comment:
...why not count them as 200 each instead of 1200 total?

Reply:
Each stat is already clamped from 0 to 200, this is just to check for any weird situations that might possibly happen. I guess I can do that though

Reply:
I wouldn't. I also wouldn't clamp it, but give an error message if it's over 200.